src/modules/cpu/kernel/log.hpp

```diff
@@ -0,0 +1,131 @@
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+#include "rppdefs.h"
+#include "rpp_cpu_common.hpp"
+
+// 1 pixel log helper functions
+inline void compute_log(Rpp8s *src, Rpp32f *dst) { *dst = log(abs((Rpp32f)(*src + 128))); }
+inline void compute_log(Rpp16f *src, Rpp16f *dst) { *dst = log(abs((Rpp16f)(*src))); }
+template<typename T>
+inline void compute_log(T *src, Rpp32f *dst) { *dst = log(abs((Rpp32f)(*src))); }
+
+// Computes ND log recursively
+template<typename T1, typename T2>
+void log_recursive(T1 *src, Rpp32u *srcStrides, T2 *dst, Rpp32u *dstStrides, Rpp32u *dstShape, Rpp32u nDim)
+{
+    if (!nDim)
+    {
+        compute_log(src, dst);
+    }
+    else
+    {
+        for (int i = 0; i < *dstShape; i++)
+        {
+            log_recursive(src, srcStrides + 1, dst, dstStrides + 1, dstShape + 1, nDim - 1);
+            dst += *dstStrides;
+            src += *srcStrides;
+        }
+    }
+}
+
+template<typename T1, typename T2>
+RppStatus log_generic_host_tensor(T1 *srcPtr,
+                                  RpptGenericDescPtr srcGenericDescPtr,
+                                  T2 *dstPtr,
+                                  RpptGenericDescPtr dstGenericDescPtr,
+                                  Rpp32u *roiTensor,
+                                  rpp::Handle& handle)
+{
+    Rpp32u numThreads = handle.GetNumThreads();
+    Rpp32u nDim = srcGenericDescPtr->numDims - 1; // Omitting batchSize here to get tensor dimension.
+    Rpp32u batchSize = dstGenericDescPtr->dims[0];
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(numThreads)
+    for(int batchCount = 0; batchCount < batchSize; batchCount++)
+    {
+        Rpp32u *roi = roiTensor + batchCount * nDim * 2;
+        Rpp32u *begin = roi;
+        Rpp32u *length = &roi[nDim];
+
+        T1 *srcPtr1 = srcPtr + batchCount * srcGenericDescPtr->strides[0];
+        T2 *dstPtr1 = dstPtr + batchCount * dstGenericDescPtr->strides[0];
+
+        for(int i = 0; i < nDim; i++)
+            srcPtr1 += begin[i] * srcGenericDescPtr->strides[i + 1];
+        if(nDim == 2)
+        {
+            T1 *srcPtrRow = srcPtr1;
+            T2 *dstPtrRow = dstPtr1;
+
+            for(int i = 0; i < length[0]; i++)
+            {
+                T1 *srcPtrTemp = srcPtrRow;
+                T2 *dstPtrTemp = dstPtrRow;
+
+                for (int vectorLoopCount = 0; vectorLoopCount < length[1]; vectorLoopCount++)
+                {
+                    compute_log(srcPtrTemp, dstPtrTemp);
```

Comment:
Please vectorized this code by using [log_ps](https://github.com/snehaa8/rpp/blob/sn/math_log_tensor/src/include/cpu/rpp_cpu_simd.hpp#L2342) function available in rpp_cpu_simd.hpp

It is an approximate version for log though, so some precision mismatches will be expected. You can try vectorizing a simple code for one such case (1D/2D/3D) and check the performance benefit we can get.

We should be able to take a call on that based on performance benefit
@r-abishek Please share your thoughts here

Reply:
log_ps() is just a one pixel approximation and I feel with load and store it might not give us the performance we intend.

Reply:
Using log_ps gives mismatch from 4th decimal

Reply:
@sampath1117 @snehaa8 We have to come up with a way to provide a vectorized implementation for a function like log. Even intel svml is proprietary so its fine to use an approximation that works up until 6 decimal places unless we have a training convergence issue that requires us to match higher.

Reply:
Added AVX optimized HOST variant and it matches with HIP upto 5 decimal places.